src/main/java/com/limechain/rpc/methods/chain/ChainRPCImpl.java

```diff
@@ -0,0 +1,192 @@
+package com.limechain.rpc.methods.chain;
+
+import com.limechain.network.protocol.warp.dto.Block;
+import com.limechain.network.protocol.warp.dto.BlockHeader;
+import com.limechain.network.protocol.warp.dto.HeaderDigest;
+import com.limechain.storage.block.BlockState;
+import com.limechain.storage.block.exception.BlockNotFoundException;
+import com.limechain.storage.block.exception.BlockStorageGenericException;
+import com.limechain.storage.block.exception.HeaderNotFoundException;
+import io.emeraldpay.polkaj.types.Hash256;
+import lombok.AllArgsConstructor;
+import org.apache.tomcat.util.buf.HexUtils;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Service;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+@Service
+@AllArgsConstructor
+public class ChainRPCImpl {
+
+    private static final String HEX_PREFIX = "0x";
+    private final BlockState blockState = BlockState.getInstance();
+
+    /**
+     * Converts an input object representing a block number into a BigInteger.
+     * The input object can be of several types, indicating the 'n-th' block in the chain:
+     * - HEX: A hex-encoded string prefixed with "0x" representing the block number.
+     * - U32: An integer or long value directly representing the block number.
+     * - ARRAY: An array of values, where each value can be either HEX or U32 type.
+     * This method is designed to handle the flexible input types and convert them into a standardized BigInteger format.
+     *
+     * @return A function that takes an Object (HEX, U32, or ARRAY) and returns a BigInteger representation of the block number.
+     * If the input is an array, the function processes each element according to its type (HEX or U32) and returns a list of BigInteger.
+     * Returns null if the input cannot be converted to a BigInteger.
+     */
+    @NotNull
+    private static Function<Object, BigInteger> parseObjectToBigInt() {
```

Comment:
It's better for private functions to be placed below the public functions they are used in, the top function of a class should never be private.